idea/idea-analysis/src/org/jetbrains/kotlin/idea/util/CommentSaver.kt

```diff
@@ -140,6 +142,7 @@ class CommentSaver(originalElements: PsiChildRange, private val saveLineBreaks:
     private val lineBreaksToRestore = ArrayList<LineBreakTreeElement>()
     private var toNewPsiElementMap by Delegates.notNull<MutableMap<TreeElement, MutableCollection<PsiElement>>>()
     private var needAdjustIndentAfterRestore = false
+    private var isSingleExpressionWithCommentBeneath = false
```

Comment:
Please convert it to a parameter of `restore`, with default value. Don't introduce additional class state when it's not required